app/main/views/suppliers.py

```diff
@@ -479,6 +479,42 @@ def set_a_declaration(supplier_id, framework_slug):
     return jsonify(declaration=supplier_framework.declaration), status_code
 
 
+@main.route('/suppliers/<int:supplier_id>/frameworks/<framework_slug>/declaration', methods=['POST'])
+def remove_a_declaration(framework_slug, supplier_id):
+    """
+    This route will replace unsuccessful applicants's declarations with an empty dict and return the altered object.
+    :param framework_slug: a string describing the framework
+    :type framework_slug: string
+    :param supplier_id: a way of identifying the supplier whose declaration will be removed
+    :type supplier_id: int
+    :return: The altered SupplierFramework object
+    :rtype: SupplierFramework
+    """
+    updater_json = validate_and_return_updater_request()
+    supplier_framework = SupplierFramework.find_by_supplier_and_framework(
+        supplier_id, framework_slug
+    ).first()
+
+    supplier_framework.declaration = {}
+
+    audit_event = AuditEvent(
+        audit_type=AuditTypes.update_supplier_framework,
+        db_object=supplier_framework,
+        user=updater_json['updated_by'],
+        data={}
+    )
+
+    db.session.add(supplier_framework, audit_event)
+    try:
+        db.session.commit()
+    except (IntegrityError, DataError):
+        db.session.rollback()
+        error_msg = "Could not remove declaration data from supplier framework: supplier_id {}, framework {}"
+        abort(400, error_msg.format(supplier_id, framework_slug))
+
+    return jsonify(message="Successful!"), 200
```

Comment:
Thought it was going to "return the altered object"? Am I missing something?

Reply:
Yes! Excellent catch, thanks Rob